hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/types/NSSummary.java

```diff
@@ -20,51 +20,67 @@
 
 import org.apache.hadoop.ozone.recon.ReconConstants;
 
+import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.List;
 
 /**
  * Class to encapsulate namespace metadata summaries from OM.
  */
 
 public class NSSummary {
   private int numOfFiles;
-  private int sizeOfFiles;
+  private long sizeOfFiles;
   private int[] fileSizeBucket;
+  private List<Long> childDir;
 
   public NSSummary() {
     this.numOfFiles = 0;
-    this.sizeOfFiles = 0;
+    this.sizeOfFiles = 0L;
     this.fileSizeBucket = new int[ReconConstants.NUM_OF_BINS];
+    this.childDir = new ArrayList<>();
   }
 
-  public NSSummary(int numOfFiles, int sizeOfFiles, int[] bucket) {
+  public NSSummary(int numOfFiles,
+                   long sizeOfFiles,
+                   int[] bucket,
+                   List<Long> childDir) {
     this.numOfFiles = numOfFiles;
     this.sizeOfFiles = sizeOfFiles;
     setFileSizeBucket(bucket);
+    setChildDir(childDir);
   }
 
   public int getNumOfFiles() {
     return numOfFiles;
   }
 
-  public int getSizeOfFiles() {
+  public long getSizeOfFiles() {
     return sizeOfFiles;
   }
 
   public int[] getFileSizeBucket() {
     return Arrays.copyOf(this.fileSizeBucket, ReconConstants.NUM_OF_BINS);
   }
 
+  public List<Long> getChildDir() {
+    return new ArrayList<>(childDir);
```

Comment:
Why do we need to create a copy of the map here? We can avoid the setter in NSSummary#setChildDir if a copy is not being created here. Is there a use case for an NSSummary object to be immutable? If yes, then there should not be a set.

Reply:
@yuangu002 mentioned a `bugcheck` complaint by just returning the array ptr itself. As the accessor can inadvertently change its content.

Reply:
I agree returning a copy of the array every time is no good for perf.

Since this is internal maybe we can suppress the bugcheck warning with an annotation.

Reply:
Instead if we really want to keep the collection internal, we could provide addChild, removeChild APIs as well. instead of setChilden(Collection)